### PR TITLE
Query updates

### LINKED ISSUE #163: Ability to customise different size modal
### Expected behaviour

The ability to provide your own HTML for the modal that renders it in different sizes based on the HTML markup. e.g.

`<div class="modal-dialog modal-lg">` renders a large modal

`<div class="modal-dialog">` renders a small model as a default

Check modal [sizes](https://getbootstrap.com/docs/3.3/javascript/#modals-sizes) for more details.

### Actual behaviour

[L68 and L69](https://github.com/linzjs/linz/blob/master/views/layout.jade#L68) of `layout.jade` was added in recent commit which limit the modal to the small size and there's no option to change it to the larger version.

### Suggested change

Remove the 2 line of codes mentioned above and allowing the user to provide a the modal markup. This way user can choose the size they want to use.

### PR DESCRIPTION
**Pull request by smebberson**
A breaking PR.

Fixes #163, provides `req` when executing the `listQuery` static, and fixes a long term development issue:

```
TypeError: filteredNavigation.forEach is not a function
    at /app/node_modules/linz/lib/api/navigation.js:66:32
    at /app/node_modules/async/dist/async.js:2549:16
    at /app/node_modules/async/dist/async.js:2488:9
    at replenish (/app/node_modules/async/dist/async.js:946:17)
    at iterateeCallback (/app/node_modules/async/dist/async.js:931:17)
    at /app/node_modules/async/dist/async.js:906:16
    at /app/node_modules/async/dist/async.js:2490:13
    at /app/node_modules/async/dist/async.js:2551:17
    at /app/node_modules/linz/lib/api/navigation.js:56:24
    at /app/node_modules/async/dist/async.js:3322:25
    at /app/node_modules/async/dist/async.js:421:16
    at iteratorCallback (/app/node_modules/async/dist/async.js:996:13)
    at /app/node_modules/async/dist/async.js:906:16
    at /app/node_modules/async/dist/async.js:3319:13
    at /app/node_modules/linz/lib/api/navigation.js:45:32
    at permissions (/app/bootstrap/linz.js:148:65)
```

### Setup

- [x] Pull down this branch.
- [x] Configure a project to use it.
- [x] Run `npm install` against Linz to get the updated version of `async`.

### Testing

- [x] Make sure you test with a model that provides the `listQuery` method.
- [x] Make sure the export modal works.
- [ ] Configure a record action, that provides `.modal-dialog .modal-content` divs. Make sure it all works as expected.

### CODE REVIEW

File: docs/models.rst [hunk 1]
```diff
@@ -250,13 +250,13 @@ listQuery static
...
-  function listQuery (query, callback)
+  function listQuery (req, query, callback)
 
-If found, Linz will execute this function with a Mongoose query before executing it, when retrieving data for the model list view. This provides an opportunity to customise the query before execution.
+If found, Linz will execute this function with ``req`` and a Mongoose query before executing it, when retrieving data for the model list view. This provides an opportunity to customise the query before execution.
 
 For example, if you'd like to return more fields from MongoDB than those listed in ``list.fields`` you can do it here::
 
-  model.static.listQuery = (query, callback) => callback(null, query.select('anotherField anotherOne'));
+  model.static.listQuery = listQuery (req, query, callback) => callback(null, query.select('anotherField anotherOne'));
```
[the remark this answers is not in the document]
Comment (smebberson) [reply docs/models.rst]: You don't need it, but actually, named functions help a lot with debugging. But I can remove it.

[the remark this answers is not in the document]
Comment (allanchau) [reply docs/models.rst]: Oh can you do that with arrow functions?

**Comment by smebberson** [comment]
@allanchau, okay, I'll take a look.

**Comment by allanchau** [comment]
@smebberson While we are here, could you also add this in to the bottom of the travis file?
```yaml
notifications:
  email:
    on_failure: never
    on_success: never
  slack:
    secure: GxIYZXaKRqfqQZ+XWj9LXHMuNfqMUsY9gcA6pm8uWhmjSkiLMWg6B6w0qLfqtCGXxCLYQDQYfJ+7mERGScr+2gDVOUWZ0DVS5DNSar4kGhQUYPskJ4MCZIFwh3ISmQYCFrl3w/rlz77OfchvkfOmOmIAIycwlhe/396Oyz2vsPo=
```

**Comment by smebberson** [comment]
@allanchau, this has all been tidied up and is working now. Back over to you.

**Comment by allanchau** [comment]
@smebberson All good, I'll leave it with you to do a release.

**Pull request merged**